File: server/CategoryWeightsServer.go
package main

import (
	context "context"
	"google.golang.org/grpc/codes"
	"google.golang.org/grpc/status"
	proto "softwareengineer-test-task/proto"
)

type CategoryWeightsServer struct {
	proto.UnimplementedCategoryWeightsServiceServer
}

func (s *CategoryWeightsServer) GetScoresByCategoriesForPeriod(context.Context, *proto.Period) (*proto.Category, error) {
	return nil, status.Errorf(codes.Unimplemented, "method GetScoresByCategoriesForPeriod not implemented")
}

func (s *CategoryWeightsServer) GetScoresByTicketsForPeriod(context.Context, *proto.Period) (*proto.Ticket, error) {
	return nil, status.Errorf(codes.Unimplemented, "method GetScoresByTicketsForPeriod not implemented")
}
func (s *CategoryWeightsServer) GetOverallScoreForPeriod(context.Context, *proto.Period) (*proto.Score, error) {
	return nil, status.Errorf(codes.Unimplemented, "method GetOverallScoreForPeriod not implemented")
}
func (s *CategoryWeightsServer) GetOverallScoreChangeForPeriodRange(context.Context, *proto.PeriodRange) (*proto.Score, error) {
	return nil, status.Errorf(codes.Unimplemented, "method GetOverallScoreChangeForPeriodRange not implemented")
}

File: Server/CategoryWeightsServer.php
<?php

namespace Server;

use CategoryWeights\Category;
use CategoryWeights\CategoryWeightsServiceStub;
use CategoryWeights\Period;
use Grpc\RpcServer;
use Grpc\ServerContext;

class CategoryWeightsServer extends CategoryWeightsServiceStub {

  public function GetScoresByCategoriesForPeriod(
    Period $request,
    ServerContext $context
  ): ?Category {
    echo "I work!";

    return null;
  }


}

$server = new RpcServer();
$server->addHttp2Port('0.0.0.0:50051');
$server->handle(new CategoryWeightsServer());
$server->run();
